semgrep_output_v1.atd

```diff
@@ -1362,6 +1364,11 @@ type finding_hashes = {
 (* CI scan results *)
 (* ----------------------------- *)
 
+type batch_info = {
```

Comment:
I talked with @akuhlens a little bit last week about this project, and it feels a bit sad to have
to support use cases where companies want semgrep to upload 10 000 findings ... what's they gonna do with
those 10000 findings? It's not a good use case; Also I don't really understand either why we need to split those findings results in batches ... it feels like we reinventing packet management and TCP. 
But, whatever.

Reply:
Yeah I agree, it's not ideal for users to be uploading 10k+ findings. But I do think that supporting batching is good in the long run since it will give the CLI and the backend more flexibility without requiring coordinated releases. 

For example, we may want the CLI to run each product in parallel and upload the results per-product. Similarly, we may want to update the prioritization of result processing, ingesting dependency results as soon as they're uploaded.

Reply:
I'm fine with your last 2 use cases, but I still don't see why we need batch_info for that ...
In fact, if what we want is streaming results, then you don't know in advance the num_batches; you'll discover this as we go.

Not a fan of reimplementing our own packet management on top of an existing packet management (TCP), which handles it far better than us. Talking to @akuhlens it seems one motivation is that we timeout sometimes because the message is too big but in that case we can just modulate the timeout in function of the size of the #findings. Also even if 10k is ridiculously high number of findings (I don't see how anybody can make use of that), it's still a small number; it's not 1 trillion finding. We should not timeout sending 10k findings ... people send huge video through the wire and TCP handles it fine. If 10k finding result in huge messages, then maybe we need to work that out. If processing 10k finding in the backend takes too long, then maybe we need to address that instead. I don't think reimplementing TCP is the solution to the problem you're trying to solve.